earthpy/spatial.py

```diff
@@ -221,6 +241,7 @@ def stack(sources, dest):
         A list with paths to the bands you wish to stack. Objects
         will be stacked in the order provided in this list.
     dest : a rio.open writable object that will store raster data.
```

Comment:
hey @joemcglinchy  this is looking really great. i suggested a change via a commit. One other question. why do you have the dest and the write_raster flag? if we are writing a raster wouldn't you just need an ouput path to assume the user wanted to write it to the hard drive? i'm just unclear as to why we need 2 arguments here but could be missing something. i'm playing with it locally and it works great so far!!

Reply:
dest and write_raster flag are both optional arguments, so the default behavior is to return an array. the flag is there as a check to make sure the user wants to write the dataset. We could instead check for a valid filename IF the user provides the filepath, but the way it is currently implemented is just an additional stage of checking that.

Reply:
ok. @joemcglinchy i think that we don't need two checks in this case. i think if the user provides a path we should 
1. check that it exists and if it exists
2. write the file 

In that instance the default value for out_path would be `None` and i'm assuming we are assuming the output driver is tiff. i could see a driver option in the future potentially i suppose.

Does that make sense to you? Or again perhaps i'm missing something here so say that if i am!!

Reply:
the file won't yet exist so i think we instead check that it is valid. We'll table to driver option for now!

Reply:
right. i'm sorry i mean the path to the file not the file itself. just that you can write to that directory succesfully. Tabling the driver option sounds perfect to me!!